lib/workers/bin/provider_worker

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../../config/environment", __dir__)
+
+ems_id = ENV["EMS_ID"]
+
+ems = ExtManagementSystem.find(ems_id)
+
+connection_config = {
+  :ems_id          => ems_id,
+  :ems_name        => ems.name,
+  :endpoints       => ems.endpoints,
+  :authentications => ems.authentications,
+  :messaging_opts  => MiqQueue.send(:messaging_client_options)
```

Comment:
TODO we should have a better way of getting this info (that isn't a private method on MiqQueue)

Reply:
I wonder if this whole thing should just be a method on ExtManagementSystem, so then this all just becomes

```ruby
connection_config = ExtManagementSystem.find(ems_id).connection_config
```

Then we can actually writes specs against it too.

Reply:
There is actually an existing method `ExtManagementSystem#connection_configurations` which returns the endpoints and authentications (I credit that for the naming here :smile:) which is great but it returns nested OpenStructs and that didn't `to_json` nicely.

Was considering converting that to return just a hash and change all the callers